frozen-abi/macro/src/lib.rs

```diff
@@ -395,19 +409,26 @@ fn frozen_abi_enum_type(input: ItemEnum, expected_digest: &str) -> TokenStream {
 #[cfg(RUSTC_WITH_SPECIALIZATION)]
 #[proc_macro_attribute]
 pub fn frozen_abi(attrs: TokenStream, item: TokenStream) -> TokenStream {
-    let args = parse_macro_input!(attrs as AttributeArgs);
     let mut expected_digest: Option<String> = None;
-    for arg in args {
-        match arg {
-            NestedMeta::Meta(Meta::NameValue(nv)) if nv.path.is_ident("digest") => {
-                if let Lit::Str(lit) = nv.lit {
-                    expected_digest = Some(lit.value());
-                }
-            }
-            _ => {}
+    let attrs_parser = syn::meta::parser(|meta| {
+        if meta.path.is_ident("digest") {
+            expected_digest = Some(meta.value()?.parse::<LitStr>()?.value());
         }
-    }
-    let expected_digest = expected_digest.expect("the required \"digest\" = ... is missing.");
+        Ok(())
+    });
```

Comment:
I am preserving existing behavior here.
But it is somewhat confusing that this macro ignores any unexpected content in the attributes.
I would expect the macro to complain about any unexpected content in its attribute.
`syn` [basic parsing example](https://docs.rs/syn/latest/syn/meta/fn.parser.html#example) shows that as well.
The downside is that a typo or other kind of error could go unnoticed, complicating the debugging process.
It could make sense to ignore existing content for the purposes of future compatibility.
But crate versioning already handles this part, does it not?

Reply:
you're correct. i was just lazy. no strong need to preserve existing behavior... 

if you have more stamina, could you make this handle errors more ergonomically?

Reply:
It is really easy - just return an error with a short message, and the `syn` library will do the rest.